django_pyscss/compiler.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import absolute_import
+
+import os
+from pathlib import PurePath
+from StringIO import StringIO
+
+from django.conf import settings
+from django.contrib.staticfiles.storage import staticfiles_storage
+
+from scss import Compiler, config
+from scss.extension.core import CoreExtension
+from scss.extension.compass import CompassExtension
+from scss.source import SourceFile
+
+from .extension.importer import DjangoImportExtension
+from .utils import find_all_files, get_file_and_storage
+
+
+# TODO: It's really gross to modify this global settings variable.
+# This is where PyScss is supposed to find the image files for making sprites.
+config.STATIC_ROOT = find_all_files
+config.STATIC_URL = staticfiles_storage.url('scss/')
+
+# This is where PyScss places the sprite files.
+config.ASSETS_ROOT = os.path.join(settings.STATIC_ROOT, 'scss', 'assets')
+# PyScss expects a trailing slash.
+config.ASSETS_URL = staticfiles_storage.url('scss/assets/')
+
+
+class DjangoScssCompiler(Compiler):
+    def __init__(self, **kwargs):
+        kwargs.setdefault('extensions', (DjangoImportExtension, CoreExtension, CompassExtension))
+        super(DjangoScssCompiler, self).__init__(**kwargs)
+
+    def compile(self, *paths):
+        compilation = self.make_compilation()
+        for path in paths:
+            path = PurePath(path)
+            if path.is_absolute():
+                path = path.relative_to('/')
+            filename, storage = get_file_and_storage(str(path))
+            with storage.open(filename) as f:
+                source = SourceFile.from_file(f, origin=path.parent, relpath=PurePath(path.name))
+            compilation.add_source(source)
+        return self.call_and_catch_errors(compilation.run)
+
+    def compile_string(self, string, filename=None):
```

Comment:
@eevee,

In order to avoid making the compressor code have to know about SourceFiles, I added the filename kwarg to compile_string.

I had to do the StringIO stuff because SourceFile.from_string ignores origin and the import code relies on origin being set.